src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,160 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import static frc.robot.Constants.ShooterConstants.flyWheelRadius;
+import static frc.robot.Constants.ShooterConstants.leftShooterNeoPort;
+import static frc.robot.Constants.ShooterConstants.rightShooterNeoPort;
+import static frc.robot.Constants.ShooterConstants.shooterD;
+import static frc.robot.Constants.ShooterConstants.shooterF;
+import static frc.robot.Constants.ShooterConstants.shooterI;
+import static frc.robot.Constants.ShooterConstants.shooterP;
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANPIDController;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.revrobotics.ControlType;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+/**
+ * Two neos spin a single flywheel shooter at incredibly high speed.
+ */
+public class Shooter extends SubsystemBase {
+
+  private static Shooter instance;
+  private CANSparkMax leftShooterNeo;
+  private CANSparkMax rightShooterNeo;
+
+  private CANPIDController PIDController;
+
+  private CANEncoder shooterEncoder;
+
+  /*These thresholds are used by commands to decide how close the flywhell's velocity needs to be to
+  its setpoint before a ball is fed.
+   */
+  public final double accurateShootVelocityThreshhold = 0.95;
+  public final double fastShootVelocityThreshold = 0.8;
+
+  //target height and shooter height in meters
+  public final double targetHeight = 5;
+  public final double shooterHeight = 4;
+  private double targetSpeed;
+
+  private Shooter() {
+    try {
+      leftShooterNeo = new CANSparkMax(leftShooterNeoPort, MotorType.kBrushless);
+      rightShooterNeo = new CANSparkMax(rightShooterNeoPort, MotorType.kBrushless);
+    } catch (RuntimeException ex) {
+      DriverStation
+          .reportError("Error Instantiating Shooter Motor Controllers: " + ex.getMessage(), true);
+    }
+    leftShooterNeo.follow(rightShooterNeo, true);
+
+    leftShooterNeo.setIdleMode(IdleMode.kCoast);
+    rightShooterNeo.setIdleMode(IdleMode.kCoast);
+
+    shooterEncoder = rightShooterNeo.getEncoder();
+    PIDController = rightShooterNeo.getPIDController();
+    PIDController.setFeedbackDevice(shooterEncoder);
+
+    targetSpeed = 0;
```

Comment:
Is it necessary to set this to 0 in the constructor vs along side declaration like the other target values?

Reply:
yes